### ISSUE TITLE
Add helper method for getting entities from a message

### ISSUE THREAD

**Issue by jacobtomlinson** [opsdroid/opsdroid#446]
# Description
Depending on which matcher is used to match a skill the `message` object has additional attributes containing the raw response from the parser.

For example if matched with `match_regex` then the `re.match` object is available at `message.regex`, if matched with `match_rasanlu` then the response object from the Rasa NLU api is available at `message.rasanlu`, etc. 

This means that if you use multiple matchers on your skill then you need to include a ton of code checking which matchers was used and then extract entities in different ways before doing anything else. You also end up duplicating that code frequently.

Example with three matchers

```python
@match_regex(r'my name is (?P<name>\w+)')
@match_rasanlu('smalltalk.name')
@match_witai('smalltalk.name')
async def my_name_is(opsdroid, config, message):
    entity_name = 'name'
    name = None
    if hasattr(message, 'regex'):
        name = message.regex.group(entity_name)
    if hasattr(message, 'rasanlu'):
        [name] = [entity["value"] for entity in message.rasanlu["entities"] if entity["entity"] == entity_name]
    if hasattr(message, 'witai'):
        name = message.witai["entities"][entity_name][0]["value"]
    if name is None:
        return False
    await message.respond("Hello {name}".format(name=name))
```

As all language matchers have the capacity to return named entities we should add a `entities` `@property` to the `Message` class which basically wraps up the if statements from the above example (along with more to cover all matchers) and extracts all entities into a dictionary.

New example

```python
@match_regex(r'my name is (?P<name>\w+)')
@match_rasanlu('smalltalk.name')
@match_witai('smalltalk.name')
async def my_name_is(opsdroid, config, message):
    entity_name = 'name'
    if entity_name not in message.entities
        return False
    await message.respond("Hello {name}".format(name=message.entities[entity_name]))
```

**Comment by jacobtomlinson** [comment]
@redfruitt have you made any progress on this?

**Comment by redfruitt** [comment]
not much yet, will buck up this weekend.

**Comment by redfruitt** [comment]
Hey @jacobtomlinson, wanted to know your thoughts on how best setup this @property within the Message class, I'm stuck a little with the regex clause.
And do you see any problems with the how I wrote rasa's entity fetch?

```python
    @property
    def entities(self):
        """Store entities created via various parsers."""
        entities = {}

        if self.regex:
            entities = [m.groupdict() for m in self.regex.finditer(self.text)]

        elif rasanlu:
            entities = rasanlu["entities"]
        return entities

```

**Comment by jacobtomlinson** [comment]
Sorry for not getting back to you @redfruitt this totally slipped off my radar. This approach did look good. 

This is up for grabs again as I imagine it has gone stale.

**Comment by jacobtomlinson** [comment]
This was closed in #955
